src/main/java/frc/robot/subsystems/Intake.java

```diff
@@ -101,20 +114,25 @@ public void toggle() {
    */
   public void setRetracted() {
     setSpinSpeed(0.0);
-    solenoid.set(DoubleSolenoid.Value.kReverse);
+    bottomSolenoid.set(DoubleSolenoid.Value.kReverse);
+    Timer.delay(IntakeConstants.INTAKE_SOLENOIDS_DELAY_TIME);
```

Comment:
Timer.delay should always be avoided. Our code is not multithreaded, so this will lock up the entire robot (which is a problem even for short times). I realize now that this poses a few problems and may not come with an obvious solution. 

I think the best way to handle this is actually to have extension-setting methods return commands that can be scheduled by the caller. This should also be done for toggle, extend, and neutral, even if it isn't needed for neutral (just to keep the pattern for these methods). The intake commands should be converted to sequential command groups that call the superconstructor with the command returned from `someIntakeSubsystem.getSomeCommand()`

Also, I think we can move the stopping of intake spin to the new command groups in the actual intake commands.

An example would be (in tightly coupled pseudocode):

```java
private Command getRetractionCommand() {
    return new SequentialGroup(
        new InstantCommand(() -> bottomSolenoid.set(reverse)),
        new WaitCommand(insert delay time here),
        new InstantCommand(() -> topSolenoid.set(reverse)),
    )
}
```

This should preserve the safety of the action (ie. don't break the robot) without making a mess of the code.

Reply:
Not sure I understand what you mean by "move the stopping of intake spin to the new command groups" - like don't do it in the getSomeCommand()?

Reply:
@kchenturtles yeah, don't include it in the command returned by getSomeCommand. In the new command group for retraction (as part of the classes that control the intake), you'd pass `new StopIntakeCommand` and `intake.getRetractionCommand()` to the superconstructor

Reply:
What about toggle? It just calls getRetractionCommand() -- should I put a stopIntakeCommand in the toggle command too?

Reply:
@kchenturtles Oh uhhhh....... Good point..... Remove the toggle logic from the subsystem and implement it in the constructor of the toggle command class. The togglecommand should check isRetracted and then use that to determine if the superconstructor should be called with a new Retract command or a new Extend command (the classes, not the methods from the subsystem).